src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,20 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+transpose1(df::AbstractDataFrame, indexcol=1) = unstack(stack(df,Not(indexcol)), :variable, indexcol,:value)
+
+function transpose2(df::AbstractDataFrame, indexcol=1)
+    m = permutedims(Matrix(df[!, Not(indexcol)]))
+    df2 = DataFrame(variable=names(df[!, Not(indexcol)]))
+    hcat(df2, DataFrame(m, df[!, indexcol]))
```

Comment:
`copycols=false` would help performance a bit here.

Above, we could also try to avoid doing two copies (one due to `Matrix` and the other due to `permutedims`), but that doesn't seem super easy if we still want to take advantage of optimized `permutedims` methods.